tests/rptest/services/flink.py

```diff
@@ -0,0 +1,214 @@
+    "taskmanager.host":
+    "localhost",
+
+    # The total process memory size for the TaskManager.
+    #
+    # Note this accounts for all memory usage within the TaskManager process, including JVM metaspace and other overhead.
+    "taskmanager.memory.process.size":
+    "1728m",
+
+    # To exclude JVM metaspace and overhead, please, use total Flink memory size instead of 'taskmanager.memory.process.size'.
+    # It is not recommended to set both 'taskmanager.memory.process.size' and Flink memory.
+    #
+    "taskmanager.memory.flink.size":
+    "1280m",
+
+    # The number of task slots that each TaskManager offers. Each slot runs one parallel pipeline.
+    "taskmanager.numberOfTaskSlots":
+    1,
+
+    # The parallelism used for programs that did not specify and other parallelism.
+    "parallelism.default":
+    1,
+
+    #==============================================================================
+    # Rest & web frontend
+    #==============================================================================
+
+    # The port to which the REST client connects to. If rest.bind-port has
+    # not been specified, then the server will bind to this port as well.
+    #
+    "rest.port":
+    8090,
+
+    # The address to which the REST client will connect to
+    "rest.address":
+    "localhost",
+
+    # Port range for the REST and web server to bind to.
+    #rest.bind-port: 8080-8090
+
+    # The address that the REST & web server binds to
+    # By default, this is localhost, which prevents the REST & web server from
+    # being able to communicate outside of the machine/container it is running on.
+    #
+    # To enable this, set the bind address to one that has access to outside-facing
+    # network interface, such as 0.0.0.0.
+    "rest.bind-address":
+    "0.0.0.0",
+
+    # Flag to specify whether job submission is enabled from the web-based
+    # runtime monitor. Uncomment to disable.
+    "web.submit.enable":
+    False,
+
+    # Flag to specify whether job cancellation is enabled from the web-based
+    # runtime monitor. Uncomment to disable.
+    "web.cancel.enable":
+    False,
+}
+
+
+class FlinkService(Service):
+    """
+    Service that runs Job and Task managers on single node
+    Jobs are separate python scripts at this point.
+    """
+    FLINK_HOME = "/opt/flink/"
+    FLINK_BIN = FLINK_HOME + "bin/"
+    FLINK_CONFIG_FILE_PATH = FLINK_HOME + "conf/flink-conf.yaml"
+    FLINK_START = FLINK_BIN + "start-cluster.sh"
+    FLINK_STOP = FLINK_BIN + "stop-cluster.sh"
+
+    def __init__(self, context, redpanda, topic, *args, **kwargs):
+        # No custom node support at this time
+        nodes_for_allocate = 1
+        # Init service
+        super(FlinkService, self).__init__(context,
+                                           num_nodes=nodes_for_allocate,
+                                           *args,
+                                           **kwargs)
+
+    def add_job(self, workload_module):
```

Comment:
can we also add the ability to run Java based flink jobs eg: https://github.com/redpanda-data/flink-repro

Some of our customers use this pattern, so it would be nice to write similar tests.

Reply:
There is a jar extension check here. And right now there is no difference really how workload made as it it copied as binary to target node. I.e. anything that flink CLI supports can be done.

Reply:
Done. Also, there must be a mechanism implemented to transfer broker addresses to java code